Add `release.yml`, update PR template

tomalec opened the pull request:
### Changes proposed in this Pull Request:


Add `release.yml`, update PR template

Introduce `changelog: none` exclusions, and new change types.
(Created with updated `yo grow` generator
https://github.com/woocommerce/grow/pull/38)


### Screenshots:

![gh rn 2](https://user-images.githubusercontent.com/17435/182618898-2c7311ba-a540-4740-88d0-a954dccb844c.gif)


### Detailed test instructions:
<!-- Add detailed instructions for how to test that this PR fixes the issue and confirm that it doesn't break any other features :) -->

1. Review PR template
2. See GH release notes generated from `trunk`, since 1.5.10: https://github.com/woocommerce/woocommerce-google-analytics-integration/releases/new?tag=1.5.14&target=trunk - choose "Previous tag" 1.5.10, click "Generate release notes"
3. Check notes generated with this config https://github.com/woocommerce/woocommerce-google-analytics-integration/releases/new?tag=1.5.14&target=add/release.yml -  choose "Previous tag" 1.5.10, click "Generate release notes"
4. Add/remove `changelog: none` to [one of the PR](https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/218)s
6. Check notes generated after the label change https://github.com/woocommerce/woocommerce-google-analytics-integration/releases/new?tag=1.5.14&target=add/release.yml -  choose "Previous tag" 1.5.10, click "Generate release notes"


### Additional details:

<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

> Dev - GH release config and the new PR template.

tomalec committed:
Add `release.yml`,

update PR template.
Introduce `changelog: none` exclusions, and new change types.
(Created with updated `yo grow` generator
https://github.com/woocommerce/grow/pull/38)

tomalec committed:
Fix grammar in the PR template

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/1613#pullrequestreview-1061441446.

puntope reviewed (commented):
⚠️  I found an error 

changelog: docs is not working:

See 

labels:
        - "changelog: docs"

against

"changelog: doc" as the label name

<img width="913" alt="Screenshot 2022-08-04 at 13 48 40" src="https://user-images.githubusercontent.com/5908855/182818189-b36c909b-ab0f-43d3-a64d-b3d3358865a4.png">


As far as I understand the expected is to group that changelog: doc into "[Doc] Documentation 📚"

puntope commented on .github/release.yml:
```diff
@@ -0,0 +1,45 @@
+
+    - title: "[Doc] Documentation 📚"
+      labels:
+        - "changelog: docs"
```
```suggestion
        - "changelog: doc"
```

puntope replied:
💅  Besides that I'd add also `type:documentation` label for this one

tomalec committed:
Fix `doc~s~` typo in release.yml

Fixes https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/223#pullrequestreview-1061606069

Co-authored-by: Miguel Pérez Pellicer <5908855+puntope@users.noreply.github.com>

tomalec committed:
Add `type: documentation` to `release.yml`,

Addresses https://github.com/woocommerce/woocommerce-google-analytics-integration/pull/223#discussion_r937591221

puntope reviewed (approved):
✅  LGTM
